homeassistant/components/emulated_hue/hue_api.py

```diff
@@ -213,25 +217,29 @@ def __init__(self, config):
         """Initialize the instance of the view."""
         self.config = config
 
-    @core.callback
-    def get(self, request, username):
+    async def get(self, request, username):
         """Process a request to get the list of available lights."""
         if not is_local(request[KEY_REAL_IP]):
             return self.json_message("only local IPs allowed", HTTP_UNAUTHORIZED)
         if username != HUE_API_USERNAME:
             return self.json(UNAUTHORIZED_USER)
 
-        json_response = {
-            "lights": create_list_of_entities(self.config, request),
-            "config": {
-                "mac": "00:00:00:00:00:00",
-                "swversion": "01003542",
-                "whitelist": {HUE_API_USERNAME: {"name": "HASS BRIDGE"}},
-                "ipaddress": f"{self.config.advertise_ip}:{self.config.advertise_port}",
-            },
-        }
+        hass = request.app["hass"]
+        return await hass.async_add_executor_job(self._build_full_states_json, request)
 
-        return self.json(json_response)
+    def _build_full_states_json(self, request):
+        """Create the json and list of entities in another job as this can take some time."""
+        return self.json(
+            {
+                "lights": create_list_of_entities(self.config, request),
```

Comment:
`create_list_of_entities` will need to be rewritten to use sync methods. However, getting entities via sync methods can be very inefficient. 

What part is taking long? How many entities are you testing with? For Websocket API we have a helper method to send json where we do the JSON serialization in a worker thread.

Reply:
`create_list_of_entities` is where the time is spent.  I only have 8 entities exposed.  There are 647 entities in the registry and I have 4 harmony hubs which seem to poll the endpoint quite freqently.

Reply:
That should not take that long 🤔 

I suggest to start with cleaning up `is_entity_exposed`. It can be a lot more aggressive in returning when it knows if something is going to be exposed or not. `entity.domain.lower()` is also unnecessary as its already lowered.

Reply:
`entity_id_to_number` is doing I/O in the event loop too, calling both `_load_json` and `save_json`…

Reply:
We should probably also just drop the TYPE_ALEXA which has been deprecated forever.

Reply:
`number_to_entity_id` also doing I/O btw

Reply:
`entity_to_json` does `hashlib.md5(entity.entity_id.encode()).hexdigest()`. That probably isn't very cheap

Reply:
<img width="1017" alt="Screen Shot 2020-03-13 at 1 33 13 AM" src="https://user-images.githubusercontent.com/663432/76595918-e0729b00-64ca-11ea-8c91-c499584de454.png">
Better.. down to 1.07% of the event loop vs 1.49% of the event loop.

It was 4+/-% before I cleaned up all the entity cruft when I had ~3000 in the registry